services/motion/builtin/move_request.go

```diff
@@ -41,78 +44,127 @@ type validatedMotionConfiguration struct {
 	angularDegsPerSec     float64
 }
 
+type requestType uint8
+
+const (
+	requestTypeUnspecified requestType = iota
+	requestTypeMoveOnGlobe
+	requestTypeMoveOnMap
+)
+
 // moveRequest is a structure that contains all the information necessary for to make a move call.
 type moveRequest struct {
+	requestType requestType
+	// origin is only set if requestType == requestTypeMoveOnGlobe
+	origin            spatialmath.GeoPose
+	logger            logging.Logger
 	config            *validatedMotionConfiguration
 	planRequest       *motionplan.PlanRequest
 	seedPlan          motionplan.Plan
 	kinematicBase     kinematicbase.KinematicBase
 	obstacleDetectors map[vision.Service][]resource.Name
 	replanCostFactor  float64
+
+	ctx               context.Context
+	cancelFn          context.CancelFunc
+	backgroundWorkers *sync.WaitGroup
+	responseChan      chan moveResponse
+	// replanners for the move request
+	// if we ever have to add additional instances we should figure out how to make this more scalable
+	position, obstacle *replanner
+	// waypointIndex tracks the waypoint we are currently executing on
+	waypointIndex *atomic.Int32
 }
 
 // plan creates a plan using the currentInputs of the robot and the moveRequest's planRequest.
-func (mr *moveRequest) plan(ctx context.Context) ([][]referenceframe.Input, error) {
-	inputs, err := mr.kinematicBase.CurrentInputs(ctx)
+func (mr *moveRequest) Plan() (state.PlanResponse, error) {
+	inputs, err := mr.kinematicBase.CurrentInputs(mr.ctx)
 	if err != nil {
-		return nil, err
+		return state.PlanResponse{}, err
 	}
 	// TODO: this is really hacky and we should figure out a better place to store this information
 	if len(mr.kinematicBase.Kinematics().DoF()) == 2 {
 		inputs = inputs[:2]
 	}
 	mr.planRequest.StartConfiguration = map[string][]referenceframe.Input{mr.kinematicBase.Kinematics().Name(): inputs}
+
 	// TODO(RSDK-5634): this should pass in mr.seedplan and the appropriate replanCostFactor once this bug is found and fixed.
-	plan, err := motionplan.Replan(ctx, mr.planRequest, nil, 0)
+	plan, err := motionplan.Replan(mr.ctx, mr.planRequest, nil, 0)
 	if err != nil {
-		return nil, err
+		return state.PlanResponse{}, err
+	}
+
+	waypoints, err := plan.GetFrameSteps(mr.kinematicBase.Kinematics().Name())
+	if err != nil {
+		return state.PlanResponse{}, err
+	}
+
+	switch mr.requestType {
+	case requestTypeMoveOnMap:
+		return state.PlanResponse{
+			Waypoints:  waypoints,
+			Motionplan: plan,
+		}, nil
+	case requestTypeMoveOnGlobe:
+		return state.PlanResponse{
+			Waypoints:  waypoints,
+			Motionplan: plan,
+		}, nil
+	case requestTypeUnspecified:
```

Comment:
I don't think theres a reason to have `requestTypeUnspecified`.  Anything that is something we dont expect will fallthrough to the error case on its on its own.